### PR TITLE
Upgrade glob to v10

### PR DESCRIPTION
**Pull request by gurgunday**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by gurgunday** [comment]
Only 1 test is failing

**Comment by gurgunday** [comment]
Windows is failing for now

File: index.js [hunk 1]
```diff
@@ -345,7 +343,7 @@ async function fastifyStatic (fastify, opts) {
       const winSeparatorRegex = new RegExp(`\\${path.win32.sep}`, 'g')
 
       for (const rootPath of Array.isArray(sendOptions.root) ? sendOptions.root : [sendOptions.root]) {
-        const files = await globPromise(path.join(rootPath, globPattern).replace(winSeparatorRegex, path.posix.sep), { nodir: true, dot: opts.serveDotFiles })
+        const files = await glob(path.join(rootPath, globPattern).replace(winSeparatorRegex, path.posix.sep), { follow: true, nodir: true, dot: opts.serveDotFiles })
```
Comment (Uzlopak) [inline index.js:346]: check what happens here, when glob is fed with the windows related options

**Comment by Uzlopak** [comment]
I ping @isaacs, maybe he has some hint on how to fix it. Maybe using posix: true?

**Comment by Uzlopak** [comment]
Time to install windows.

**Comment by Uzlopak** [comment]
@gurgunday 
Can you come online on discord, please.

**Comment by Fdawgs** [comment]
This isn't possible as glob dropped support for node 14 in v9.x and we're still supporting node 14. :(

**Comment by Uzlopak** [comment]
node 14 tests pass so far and this has to be a semver major anyway.

**Comment by gurgunday** [comment]
Yeah even after purging Windows from all my machines, it seems to still cause me trouble 🤠

Without an educated comment on why things aren't working, I can only keep commit-guessing 🤣

**Comment by gurgunday** [comment]
I'll try again on a Windows machine

**Pull request closed without merging**